doc/eng_sys_checks.md

```diff
@@ -0,0 +1,200 @@
+`azure-core`
+`azure-eventhub`
+`azure-storage-blob`
+ 
+Our regression framework automatically finds any such package that is added as required package so this list is not hardcoded. 
+ 
+We have two different set of regression tests to verify regression scenarios against oldest and latest released dependent packages. 
+•   Regression using latest released dependent package 
+•   Regression using oldest released dependent package 
+ 
+One main difference between regression tests and forward dependency test( latest and mindependency) is in terms of what test cases are executed as part of the tests. While forward dependency tests executes the test cases in current code repo, regression tests execute the tests that were part of repo at the time of dependent package release. To make it more clear, let's look at an example here. 
+ 
+Let's assume that we are testing regression for azure-core and this test is for regression against latest released dependent packages. Test will identify all packages that takes azure-core as required package and finds latest released version of those packages. Test framework install currently being developed azure-core and latest released dependent package and runs the test cases in dependent package, for e.g. azure-identity, that were part of repo at the time of releasing depending package. 
+ 
+Workflow of this test is as follows when running regression for an SDK package. 
+1.  Identify any packages that takes currently being tested package as required package 
+2.  Find latest and oldest released versions of dependent package from PyPI 
+3.  Install currently being developed version of package  we are testing regression for. E.g. azure-core 
+4.  Checkout the release tag of dependent package from github 
+5.  Install latest/oldest version of dependent package. For e.g. azure-identity 
+6.  Run test cases within dependent package from checked out branch.  
+ 
+ 
+Steps to manually run regression test locally: 
+1.  Run below command from your git code repo to generate the wheel of package being developed. Currently we have restricted to have prebuilt wheel. 
+`./scripts/devops_tasks/build_packages.py --service= <service-name> -d <out-folder>`
+2.  Run below command to start regression test locally 
+`./scripts/devops_tasks/test_regression.py azure-* --service=<service-name> --whl-dir=<out-folder given above in step 2>`
+ 
+ 
+How to run these additional tests on azure pipelines manually 
+ 
+Following variables can be set at queueing time in order to run these additional tests which are by default run only for scheduled runs. 
+ 
+•   Latest and oldest dependency test in addition to basic testing 
+Variable name: `Run.DependencyTest`
+Value: true 
+ 
+•   Regression test 
+Variable name: `Run.Regression`
+Value: true 
+
+
+
+## Release Checks
+
+<will be updated soon>
```

Comment:
Should we leave it this for now. or keeping it as coming soon.

Reply:
Man I'm dumb. I actually thought I cleaned that up. *screaming*